server/perkeepd/ui/blob_item_audio_content.js

```diff
@@ -0,0 +1,178 @@
+/*
+Copyright 2019 The Perkeep Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+goog.provide('cam.BlobItemAudioContent');
+
+goog.require('goog.math.Size');
+
+// Renders audio blob items.
+cam.BlobItemAudioContent = React.createClass({
+	displayName: 'BlobItemAudioContent',
+
+	audioRef_: null,
+
+	propTypes: {
+		blobref: React.PropTypes.string.isRequired,
+		filename: React.PropTypes.string.isRequired,
+		href: React.PropTypes.string.isRequired,
+		size: React.PropTypes.instanceOf(goog.math.Size).isRequired,
+	},
+
+	getInitialState: function() {
+		return {
+			mouseover: false,
+			playing: false,
```

Comment:
@mpl in response to your comments on Gerrit:

>I think the point was not about the media being loaded, but about the react element itself. imagine there are 50 videos in your current page, then it probably takes a little bit of time for all the buttons to load on the page.
>[Source](https://perkeep-review.googlesource.com/c/perkeep/+/19326/1/server/perkeepd/ui/blob_item_audio_content.js#37)

Since we're using React, buttons are created and added to the DOM on the client-side. It's a safe assumption that your buttons will be rendered soon after your React component is instantiated.

Correction, I said `loaded` is not used in video preview, it is, here is how `loaded` is used in `blob_item_video_content`:
* Adds the class `cam-blobitem-video-loaded` if `loaded`, however, class is not implemented
* Renders `getPoster_` if not `loaded`
* Renders `getVideo_` if `loaded`
* Sets `loaded` to `true` when clicking play or pause button

Basically, loaded is whether or not the video was ever played by the user. We use that information to decide whether to display the poster or the video. Audio has no visual change when it is being played, so we don't need `loaded` here.

Reply:
understood.